File: vs/CSharpCompiler/Lab1/Lab1_task5_shared.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1
{
    public class Lab1_task5_shared
    {
        public static List<string>Scan(string input)
        {
            string[] months =new string [12]{"января","февраля","марта","апреля","мая","июня",
            "июля","августа","сентября","октября","ноября","декабря"};
            int mlen = months[0].Length;
            for(int j=1;j<months.Length;j++)
                if (months[j].Length > mlen)
                    mlen = months[j].Length;
            string cesh = "";
            List<string> result = new List<string>();
            int pos = 0;
            
            string month_ = "";
            bool suc = false;
            int flag1 = 0;
            foreach (char _char_ in input.ToLower().ToCharArray())
            {
                if (pos == 0)
                {
                    if(_char_>='0'&& _char_ <= '9')
                    {
                        cesh = _char_.ToString();
                        pos = 1;
                    }
                    else
                    {
                        pos = 0;
                        cesh = "";
                        
                    }
                }
                else
                {
                    if (pos <3)
                    {
                        if(_char_==' ')
                        {
                            pos = 3;
                            suc = false;
                            cesh+= _char_.ToString();
                        }
                        else
                        {
                            if (_char_ >= '0' && _char_ <= '9')
                            {
                                cesh+= _char_.ToString();
                                pos = 2;
                            }
                            else
                            {
                                pos = 0;
                                cesh = "";
                               
                            }
                        }
                    }
                    else
                    {
                        if (flag1 == 0)
                        {
                            if (_char_ != ' ')
                            {
                                if (month_.Length <= mlen)
                                {
                                    cesh += _char_.ToString();
                                    month_ += _char_.ToString();
                                    suc = false;
                                    foreach (string _m in months)
                                        if (_m.Contains(month_))
                                        {
                                            if (_m.Length >= month_.Length)
                                            {
                                                suc = true;
                                                break;
                                            }
                                        }
                                    if (!suc)
                                    {
                                        cesh = "";
                                        
                                        month_ = "";
                                        pos = 0;
                                    }
                                }
                            }
                            else
                            {
                                cesh += _char_.ToString();
                                flag1 = 1;
                            }
                        }
                        else
                        {
                            if (_char_ >= '0' && _char_ <= '9')
                            {
                                flag1 = 2;
                                cesh+= _char_.ToString();

                            }
                            else
                            {
                                if (flag1 == 2)
                                {
                                    result.Add(cesh);
                                    pos = 0;
                                    cesh = "";
                                    month_ = "";
                                    flag1 = 0;
                                }
                                else
                                {
                                    pos = 0;
                                    cesh = "";
                                    month_ = "";
                                    flag1 = 0;
                                }
                            }
                        }
                    }
                }
            }
            return result;
        }
    }
}

File: vs/PythonApplications/Lab1_task5_shared.py

#Дана строка. Необходимо найти все даты, которые описаны в 
#виде "31 февраля 2007".
import re

def scan(input_str):
    months = ["января", "февраля", "марта", "апреля", "мая", "июня",
              "июля", "августа", "сентября", "октября", "ноября", "декабря"]
    max_len = max(len(month) for month in months)
    result = []
    pos = 0
    month = ""
    success = False
    flag1 = 0
    for char in input_str.lower():
        if pos == 0:
            if char.isdigit():
                cesh = char
                pos = 1
            else:
                pos = 0
                cesh = ""
        else:
            if pos < 3:
                if char == " ":
                    pos = 3
                    success = False
                    cesh += char
                else:
                    if char.isdigit():
                        cesh += char
                        pos = 2
                    else:
                        pos = 0
                        cesh = ""
            else:
                if flag1 == 0:
                    if char != " ":
                        if len(month) <= max_len:
                            cesh += char
                            month += char
                            success = False
                            for m in months:
                                if m.startswith(month):
                                    if len(m) >= len(month):
                                        success = True
                                        break
                            if not success:
                                cesh = ""
                                month = ""
                                pos = 0
                    else:
                        cesh += char
                        flag1 = 1
                else:
                    if char.isdigit():
                        flag1 = 2
                        cesh += char
                    else:
                        if flag1 == 2:
                            result.append(cesh)
                            pos = 0
                            cesh = ""
                            month = ""
                            flag1 = 0
                        else:
                            pos = 0
                            cesh = ""
                            month = ""
                            flag1 = 0
    return result





s='''
"123 asdfgiiIIII  34 февраля 2021 года , 11 мая,   ________________" +
                "\n\n 2 Мая 1998-го года ,,,,,,,,,,," +
                "ууууууууууууууууrrrrYYYYYY11 июня 2011////" +
                "09 июня 2001 года, 11 Апрелят 2000 ------" +
                "111августа 2022 , ЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮЮ" +
                "-1 сентября 2011 года"
'''
r=scan(s)
print(r)